### PR TITLE
Added scheduled jobs

### PR DESCRIPTION
**Pull request by nlwstein**
#### Summary of changes

**Asana Ticket:** [Setup GitHub Actions for nightly CI + dev deploy runs](https://app.asana.com/0/584764604969369/1203566390446223/f)

This adds a build and dev deploy workflow run every day at midnight. 

#### Reviewer Checklist

- [ ] Meets ticket's acceptance criteria
- [ ] Tests were added for any new functionality
- [ ] [GTFS documentation](https://github.com/mbta/gtfs-documentation/) has a PR (if data is added/changed)
- [ ] Relevant stakeholders are notified (if data is changed)
- [ ] Changes are verified on schedule_viewer or PartnerDash

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: Deploy to Dev
 
 on:
+  schedule:
+    - cron: '0 5 * * *'
```
[the remark this answers is not in the document]
Comment (bklebe) [reply .github/workflows/dev.yml:5]: @nlwstein @bfauble for now, could we do the automated deploy to dev-green and leave dev for manual testing per Brian? per the RFC https://github.com/mbta/technology-docs/blob/main/rfcs/accepted/0015-consistent-environment-naming.md, 

> The main implementation step that will be needed is that applications with existing `dev` environments will need to rename them to `staging`.

> `staging`: used for final integration testing and monitoring for errors before deploying changes to production. Often, merging a PR into the `main` / `master` branch will be set up to automatically initiate a deploy to the application's staging environment. There should be one stage environment per application.

Longer term, it seems like these nightly deploys would be better suited to `test`:

> `test`: used as the target environment for some sort of recurring automated testing, such as end-to-end web application testing or load testing. Accepts an optional modifier specifying what testing tool the environment is for (for instance, `test-cypress` or `test-locust`). This is presently a hypothetical use case, albeit one that has been discussed from time to time.

We don't currently have a `test` environment, but we should probably add one for this purpose. Could you make a ticket?

**Review by bklebe — CHANGES_REQUESTED** [review]
@nlwstein, please move the scheduled cron deploy to `dev-green` for now.

File: .github/workflows/dev.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nlwstein) [reply .github/workflows/dev.yml:5]: @bklebe That makes sense. Should I follow this rough pattern on the other PRs?

[the remark this answers is not in the document]
Comment (bklebe) [reply .github/workflows/dev.yml:5]: Yes, that would be great. Thank you.

**Comment by nlwstein** [comment]
> @nlwstein, please move the scheduled cron deploy to `dev-green` for now.

All set - https://github.com/mbta/concentrate/pull/264/commits/46daa9a95f61f46680ab0d568ab3d0d67c852448

**Review by bklebe — APPROVED** [review]

**Pull request merged**